### DISCUSSION TITLE
Variable Compression or Selective Encoding?

### DISCUSSION THREAD

**Discussion by edent** [ImageMagick/ImageMagick#6376] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

Ubuntu 22.04

### Description

I have a JPEG. I want the top half of the image to have a quality of 75% and the bottom half to be 100%.

Is it possible to change the compression part way through an image?

Do any image formats support variable compression?

(I'm very happy to be told that this is a stupid idea!)

**Comment by snibgo** [comment]
As far as I know, an image in a JPEG file can have only one compression (`-quality`) setting.

IM does not support variable compression settings for any image format.

We can simulate the effect of two (or more) compression settings, by breaking the image into two crops, saving then as JPEG files with different compressions, reading those in, appending them, and saving them in any lossless format.
```
magick toes.png -crop 100x50% +repage ( -clone 0 -quality 10 +write x0.jpg +delete ) -delete 0 -quality 90 x1.jpg

magick x0.jpg x1.jpg -append x.png
```
The result, x.png, is:

![x](https://github.com/ImageMagick/ImageMagick/assets/33812368/28347c45-6305-48da-a753-ac309292c7a9)
